src/opm/output/eclipse/AggregateMSWData.cpp

```diff
@@ -610,6 +626,72 @@ namespace {
                 usys.from_si(M::length, sicd.length());
         }
 
+        template <class RSegArray>
+        void assignAICDCharacteristics(const ::Opm::Segment&    segment,
+                                            const ::Opm::UnitSystem& usys,
+                                            const int                baseIndex,
+                                            RSegArray&               rSeg)
+        {
+            using Ix = ::Opm::RestartIO::Helpers::VectorItems::RSeg::index;
+            using M  = ::Opm::UnitSystem::measure;
+
+            const auto& aicd = segment.autoICD();
+
+            rSeg[baseIndex + Ix::DeviceBaseStrength] =
+                usys.from_si(M::aicd_strength, aicd.strength());
+
+            rSeg[baseIndex + Ix::ScalingFactor] = ((aicd.methodFlowScaling() == 1) ||
+             ((aicd.methodFlowScaling() < 0) && (aicd.length() < 0))) ?
+                usys.from_si(M::length, aicd.scalingFactor()) : aicd.scalingFactor();
+
+            rSeg[baseIndex + Ix::CalibrFluidDensity] =
+                usys.from_si(M::density, aicd.densityCalibration());
+
+            rSeg[baseIndex + Ix::CalibrFluidViscosity] =
+                usys.from_si(M::viscosity, aicd.viscosityCalibration());
+
+            rSeg[baseIndex + Ix::CriticalWaterFraction] = aicd.criticalValue();
+
+            rSeg[baseIndex + Ix::TransitionRegWidth] =
+                aicd.widthTransitionRegion();
+
+            rSeg[baseIndex + Ix::MaxEmulsionRatio] =
+                aicd.maxViscosityRatio();
+
+            rSeg[baseIndex + Ix::FlowRateExponent] =
+                aicd.flowRateExponent();
+
+            rSeg[baseIndex + Ix::ViscFuncExponent] =
+                aicd.viscExponent();
+
+            rSeg[baseIndex + Ix::MaxValidFlowRate] =
+            (aicd.maxAbsoluteRate() == std::numeric_limits<double>::max()) ?
```

Comment:
Hmm floating point `==` comparison?

Reply:
The reason for this test is that Eclipse uses a different value (-2.e+20) for the case when no maximum limit has been assigned. Flow then uses this maximum numeric limit value, which I have used for comparison - I do not know a different way to identify this special case. Please advice a different approach, I am happy to change the code!

Reply:
My main point was that value `-2e+20` should have a name - I guess the value `1e+20` is defined as a max somewhere? For the rate my suggestion would be to change[1] the `SICD::maxAbsoluteRate()` to return a `std::optional<double>` - then the whole limit value can be avoided:
```
const auto& max_rate  = aicd.maxAbsoluteRate();
if (max_rate.has_value())
      rSeg[baseIndex + Ix::MaxValidFlowRate] = usys.from_si( M::geometric_volume_rate, max_rate.valuie());
else 
     rSeg[] = -2 * EclipseInfinity;
```

Reply:
Thanks for good suggestion, I have changed the code as suggested!